### PR TITLE
Header 100 restyle

### PR DESCRIPTION
**Pull request by devin5**
I realize this wasn't an ideal-sized PR it's slightly large. There was a refactoring of the animation hook for the header and once changed all instances of transition hook were also changed. I think it's a good call to get these in together though. they touch a lot of the same places += we are in crunch time.  Header and Breadcrumb are looking great, the code is much cleaner and the animation styling objects are more readable and not so hard to follow. No bugs or known updates needed.
## Page 7 in New UI submitted as of May 21, 2021

https://www.figma.com/file/gpB87FcryeAf14le3YqbFV/rocketsoft-okr?node-id=409%3A2646

-updates styles in the header
-updates styles in the breadcrumb
-refactors the animation hook to use reducer and use a more robust, and scalable pattern
-unloads some of the formatting and mutation handling to _Lodash
-any component touched was broken out into a newly created file hierarchy to match the Facebook teams react style sheet
-adds a container wrapper to make animations more consistent

## SPEC

Screen Shot 2021-05-23 at 4 09 57 PM

## UPDATED V4

Screen Shot 2021-05-23 at 4 15 55 PM
90-=

### CODE REVIEW

File: src/components/shared/Header/BuildHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Typography as T } from "@material-ui/core";
+
+const BuildHeader = (txt, variantStr, classes) => {
+  return (
+    <T
```
Comment (danchapman) [inline src/components/shared/Header/BuildHeader.jsx:6]: What's the reason for using `T` instead of `Typography`

File: src/hooks/utils/reduceStylesObject.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const makeReduceStylesObject = (params, ...args) => {
+  const {start, end} = { start: cloneDeep(params.classesStart), end: cloneDeep(params.classesEnd) }
+  args = [...args].map(item => Array.isArray(item) ? clone(item) : item);
+
+  function reduceStylesObject() {
+    //https://www.codecademy.com/forum_questions/51068e93f73ad4947a005629   ??TODO ASK Chappy about setting a new object everytime like this
+    return reduce(
+      [...args],
+      (acc, curr) =>
+        new Object({
```
Comment (danchapman) [inline src/hooks/utils/reduceStylesObject.js:12]: I think an object literal would work fine here. We can talk about how the overall arch here seems a little complicated.

**Review by danchapman — CHANGES_REQUESTED** [review]

**Comment by danchapman** [comment]
`src/components/homegridpage` is an empty file and needs to be removed.

File: src/hooks/utils/reduceStylesObject.js (hunk 2 above)
Comment (devin5) [reply src/hooks/utils/reduceStylesObject.js:12]: expected return. thanks for lesson.
